### DISCUSSION TITLE
(Submod) Dreams and Nightmares

### DISCUSSION THREAD

**Discussion by wboorman** [Monika-After-Story/MonikaModDev#7820] [general - submod / external content]
Monika After Story:  Dreams and Nightmares  ([link](https://github.com/DrakeTheDuelist/MAS-Dreams-and-Nightmares))

Description:  An episodic branching narrative;  Monika can have dreams and tell you about them, but not all dreams are happy ones...

I'd like to get my submod added to the list of recognized submods and have it technically looked over.  A previous independent deployment attempt blew up _horribly_ and I'd like to avoid any major issues like that moving forward.

This is less of a technical submod and more of a narrative one (essentially being a giant flowchart of discussion topics that progressively unlock), so character input can also be taken into consideration, including but not limited to how Monika would phrase certain lines.  Plus, if you can think of any other persistent variables I could use to more thoroughly customize Monika's behaviors to her player, let me know.

Also I'd like some stylistic feedback.  I'm mostly coming at MAS submod development from a more professional background that often leans heavily on OOP, but otherwise I'm basically a python / renpy noob.

EDIT:  I'm also trying to put the "submod" label on this post, but for the life of me I can't figure out where that setting can be set...

**Comment by wboorman** [comment]
Well it's not ready yet...  I don't even have a release set up yet, and even then there's so much more I'd like to do.

**Comment by GlitchyMoni** [comment]
> Well it's not ready yet... I don't even have a release set up yet, and even then there's so much more I'd like to do.

An error appears.

I'm sorry, but errors were detected in your script. Please correct the
errors listed below, and try again.


File "game/Submods/DaN-_devutils.rpy", line 46: expected statement.
    © 2021 GitHub, Inc.
                 ^

Ren'Py Version: Ren'Py 6.99.12.4.2187

**Comment by lunulae** [comment]
@GlitchyMoni I would be careful with installing this before it's finished, unless you have a separate dev save for testing.

@wboorman I'm not so good on the coding side of things but when I'm finished with some of my current projects for the core mod I'd be happy to help review dialogue, if that's something needed. Excited to see more submod projects in the works!

**Comment by GlitchyMoni** [comment]
> @GlitchyMoni I would be careful with installing this before it's finished, unless you have a separate dev save for testing.
> 
> @wboorman I'm not so good on the coding side of things but when I'm finished with some of my current projects for the core mod I'd be happy to help review dialogue, if that's something needed. Excited to see more submod projects in the works!

Oh okay. Thanks for the advice!

**Comment by wboorman** [comment]
It's also probably not helped by the fact that I'm literally messing with the code right now, and the only reason I'm pushing anything is for the source control to save what I'm doing (partially catastrophe insurance, partially migrating my work to my desktop).  I make no stability guarantees until I have a nice and shiny 1.0 release.

**Comment by YoraeRasante** [comment]
Hmm... curious. This description reminds me of Tsukihime's sequel Kagetsu Tohya, where it becomes clear things are a bit too dream-like pretty quickly, as well a Fate Stay Night's similar sequel Fate/Hollow Ataraxia. Same creator, by the way, for those Fate fans.
Which is a nice deal for me, because reading the fantranslation of Tsukihime, and afterwards KT, was what made me interested in visual novels, which one day brought me to DDLC and here.

What I am saying is I'll love to try it when it becomes more stable.

**Comment by xXMidnight** [comment]
wait- so it's not usable yet? Darn. I just got a traceback text given to me, and I had to remove the submod. ;-; I was looking forward to this...
